sqlmesh/core/engine_adapter/clickhouse.py

```diff
@@ -296,6 +299,73 @@ def _build_partitioned_by_exp(
             this=exp.Schema(expressions=partitioned_by),
         )
 
+    def add_nulls_after_join_setting(
```

Comment:
[Nit] Now the name is too prescriptive and specific to clickhouse. For example, I can see how the same behavior can be achieved using query modifications other than a setting potentially.


A more descriptive name could be something like `ensure_nulls_for_unmatched_after_join`